File: samples/python-celery-reports/apphost.cs
#:package Aspire.Hosting.Python@9.2.1
#:package Aspire.Hosting.Redis@9.2.1
#:sdk Aspire.AppHost.Sdk@9.2.1

var builder = DistributedApplication.CreateBuilder(args);

var redis = builder.AddRedis("broker");

builder.AddPythonApp("api", "./src", "main.py")
    .WithReference(redis)
    .WithHttpEndpoint(env: "PORT")
    .WithExternalHttpEndpoints();

builder.AddPythonApp("worker", "./src", "worker.py")
    .WithReference(redis);

builder.Build().Run();

File: samples/python-celery-reports/apphost.py
import sys
from pathlib import Path
sys.path.insert(0, str(Path(__file__).parent / ".modules"))

from aspire import create_builder

builder = create_builder()

redis = builder.add_redis("broker")

builder.add_python_app("api", "./src", "main.py") \
    .with_reference(redis) \
    .with_http_endpoint(env="PORT") \
    .with_external_http_endpoints()

builder.add_python_app("worker", "./src", "worker.py") \
    .with_reference(redis)

app = builder.build()
app.run()
